src/firewheel/tests/unit/control/test_repository_db.py

```diff
@@ -25,6 +25,8 @@ def repository_db():
     repository_db = RepositoryDb(
         db_filename="test_repositories.json",
     )
+    for repo in repository_db.list_repositories():
+        repository_db.delete_repository(repo)
     yield repository_db
```

Comment:
Minor suggestion (as discussed), potentially we move the cleanup after the yield to clearly identify this as a "cleanup" action. More information can be found in the [pytest docs](https://docs.pytest.org/en/6.2.x/fixture.html#teardown-cleanup-aka-fixture-finalization) about this process of "fixture finalization".

```suggestion
    yield repository_db
    # Ensure all repositories are removed during teardown
    for repo in repository_db.list_repositories():
        repository_db.delete_repository(repo)
```